### PR TITLE
Check result of memory allocation for stickified data

### PR DESCRIPTION
**Pull request by mikeessen**
This PR updates code to check the result of memory allocation for stickified data.

### CODE REVIEW

File: src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp [hunk 1]
```diff
@@ -200,6 +200,7 @@ Value insertAllocOrEmitZeroConstant(ArrayRef<IndexExpr> dims,
     int64_t sizeInBytes =
         affine::getIntOrFloatMemRefSizeInBytes(resType).value();
     char *rawData = (char *)malloc(sizeInBytes);
+    assert(rawData && "failed to allocate memory for stickified data");
```
Comment (AlexandreEichenberger) [inline src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp:203]: good idea, can probably systematically add asserts wherever the compiler use `malloc`

Comment (mikeessen) [reply src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp:203]: Thanks!  Added some additional asserts

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, getting closer to trap these segfaults, thanks Mike.

**Comment by mikeessen** [comment]
> LGTM, getting closer to trap these segfaults, thanks Mike.

Thanks Alex!

**Pull request merged**